src/utils/parser/validate.js

```diff
@@ -0,0 +1,12 @@
+// @flow
+
+import { parseExpression } from "./utils/ast";
+
+export function hasSyntaxError(input: string) {
+  try {
+    parseExpression(input);
+    return false;
+  } catch (e) {
+    return `${e.name} : ${input} is an invalid.`;
```

Comment:
hmm, it might be nice to do `${e.name} : ${e.message}`;

This would communicate the syntax error which is more descriptive.

Reply:
hahahaha ... changed it based on your suggestion in a previous review :)